Changelog.md

```diff
@@ -1,5 +1,9 @@
 ### 0.8.16 (unreleased)
 
+Important Bugfixes:
+ * Code Generation: Fix a data corruption: When abi-encoding a dynamic array followed by a static calldata array the length of the dynamic array would get overwritten by ``0``.
```

Comment:
Can we make this more specific? Why has it not occurred earlier?

Reply:
It existed for a while, my guess is the combination just wasn't used enough that people reported the problem?

I rephrased it to 
` * Code Generation: Fix a data corruption: When abi-encoding a dynamic type followed by a static calldata array, the cleanup of the static array would mistakenly zero out the first slot of the of tail-part of the dynamic type that follows it.`
now

Reply:
It's even more specific. Based on what I suggested `bugs.json` I'd describe it like this:
`* Code Generation: Fix corruption of the leading 32 bytes of the first dynamically-sized component of an ABI-encoded tuple when the last component contained a statically-sized calldata array.`

Or maybe this - more verbose but probably easier for users to figure out in what circumstances it happens:
`* Code Generation: Fix data corruption that affected ABI-encoding of calldata values represented by tuples: structs at any nesting level; argument lists of external functions, events and errors; return value lists of external functions. Up to 32 leading bytes of the first dynamically-sized value in the tuple would get zeroed when the last component contained a statically-sized array.`